internal/api/keppel/accounts.go

```diff
@@ -227,10 +236,22 @@ func parseRBACPolicy(policy RBACPolicy) (keppel.RBACPolicy, error) {
 		RepositoryPattern: policy.RepositoryPattern,
 		UserNamePattern:   policy.UserNamePattern,
 	}
+	// validate cidr early to prevent errors
+	// this has also the nice side effect that we can use the cidr of the network incase an ip is used
+	if cidr := policy.CidrPattern; cidr != "" {
+		_, net, err := net.ParseCIDR(cidr)
+		if err != nil {
+			// err.Error() sadly does not contain any useful information why the cidr is invalid
+			return result, fmt.Errorf("%q is not a valid cidr", cidr)
+		}
+		result.CidrPattern = net.String()
```

Comment:
When rendering existing policies, we treat `0.0.0.0/0` as "unset", but when taking in new policies, we don't apply the same rigor. I think the minimal required change is:

```suggestion
		result.CidrPattern = net.String()
		if result.CidrPattern == "0.0.0.0/0" {
			result.CidrPattern = ""
		}
```

Also, some test coverage would be useful. At least, having a policy with only `match_cidr = "0.0.0.0/0"` and some permissions should cause the complaint that no matches are set at all.